### PR TITLE
Gives better feedback when control or alt-clicking pressure pumps, volume pumps, etc.

### LINKED ISSUE #15400: Ctrl+Clicking and alt+Clicking atmos pumps doesn't give correct feedback
**Issue Description**:

You can ctrl+click and alt+click on a valve (pumps, pressure pumps, air filters, etc) to turn it on/off and max its output. These functions work correctly, but you either get no chat notification or you get the "you can't pull the pump cause it's attached to the floor" chat line.

**What did you expect to happen**: 

Get correct feedback like "You turn the pressure valve on/off" and "You max the output on the pressure valve".

**What happened instead**:

"You can't pull the valve you silly"

**Why is this bad/What are the consequences**:

Obscure keyboard shortcuts that are inmensely helpful for making atmos (and now the SM) less annoying to set up should be easier to discover or learn about and should have appropriate chat feedback when used

**Steps to reproduce the problem**:

Just ctrl+click or alt+click a valve

**When did the problem start happening**:

Probably a long time ago? Might be ever since the functionality was added in #11348

**Extra information**:

### PR DESCRIPTION
**Pull request by SteelSlayer**
## What Does This PR Do

1. Removes the "It's anchored to the floor!" messages when you control-click on pressure pumps, volume pumps, gas filters, and gas mixers. Adds a new chat message for this: "You toggle X on/off".
2. Adds a new chat message for alt-clicking these devices: "You set the target pressure of X to maximum."
3. Small refactor of some variables. I had to unfortunately change a bunch of files because tons of atmos machinery were declaring `var/on` and `var/target_pressure` all over the place. Made these base vars on `/obj/machinery/atmospherics`

## Why It's Good For The Game
Better, and correct feedback to players.

Fixes #15400

## Changelog
:cl:
tweak: Removes the "It's anchored to the floor!" messages when you control-click on pressure pumps, volume pumps, gas filters, and gas mixers
tweak: Adds feedback messages when alt or control-clicking on pressure pumps, volume pumps, gas filters, and gas mixers
/:cl:

### CODE REVIEW

File: code/ATMOSPHERICS/components/trinary_devices/mixer.dm [hunk 1]
```diff
@@ -6,37 +6,29 @@
...
-		return
-	if(!in_range(src, user) && !issilicon(usr))
-		return
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	set_max()
-	return
+	if(can_use_shortcut(user))
+		set_max(user)
+	return ..()
```
Comment (dearmochi) [inline code/ATMOSPHERICS/components/trinary_devices/mixer.dm:31]: Some of the other machineries modified in this PR don't include this call to parent -- I'd probably include it as `/obj/machinery/atmospherics/AltClick()` handles vent crawling.

**Review by dearmochi — COMMENTED** [review]
Just one thing

File: code/ATMOSPHERICS/components/trinary_devices/mixer.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/ATMOSPHERICS/components/trinary_devices/mixer.dm:31]: I suppose but, should we call parent here even though the mixer, filter, and pumps don't allow you ventcrawl into them?

Comment (dearmochi) [reply code/ATMOSPHERICS/components/trinary_devices/mixer.dm:31]: If we go up the parents...

https://github.com/ParadiseSS13/Paradise/blob/1d891384364a64f99948658d5124c92cb6e547e7/code/ATMOSPHERICS/atmospherics.dm#L295-L299

https://github.com/ParadiseSS13/Paradise/blob/1d891384364a64f99948658d5124c92cb6e547e7/code/_onclick/click.dm#L334-L343

Turns out alt-clicking would just pass the call to the parent to the parent if not a valid vent crawl object, showing the turf contents on the stat panel. Probably don't want that as it would ruin the flow of the shortcut -- feel free to ignore my comment above.

**Review by dearmochi — APPROVED** [review]

File: code/ATMOSPHERICS/components/trinary_devices/mixer.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/ATMOSPHERICS/components/trinary_devices/mixer.dm:31]: Yeah, and in fact I should probably remove this `..()`, don't know why I added it here, but nowhere else.

**Review by AffectedArc07 — APPROVED** [review]

File: code/game/machinery/machinery.dm [hunk 2]
```diff
@@ -650,3 +650,19 @@ Class Procs:
...
+ *
+ * Arguments:
+ * * user - the mob who is trying to interact with the machine.
+ */
+/obj/machinery/proc/can_use_shortcut(mob/living/user)
+	if(!ishuman(user) && !issilicon(user))
+		return FALSE
+	if(!in_range(src, user) && !issilicon(user))
+		return FALSE
+	if(user.incapacitated())
```
Comment (SabreML) [inline code/game/machinery/machinery.dm:665]: This instead maybe, to avoid two `issilicon()` checks?
```suggestion
/obj/machinery/proc/can_use_shortcut(mob/living/user)
	if(ishuman(user))
		if(!in_range(src, user))
			return FALSE
	else if(!issilicon(user))
		return FALSE
	if(user.incapacitated())
```

File: code/modules/atmospherics/machinery/atmospherics.dm [hunk 3]
```diff
@@ -353,3 +356,33 @@ Pipelines + Other Objects -> Pipe network
...
+ *
+ * Arguments:
+ * * user - the mob who is toggling the machine.
+ */
+/obj/machinery/atmospherics/proc/toggle(mob/living/user)
+	if(powered())
+		on = !on
+		update_icon()
+	if(user)
+		to_chat(user, "<span class='notice'>You toggle [src] [on ? "on" : "off"].</span>")
```
Comment (SabreML) [inline code/modules/atmospherics/machinery/atmospherics.dm:373]: Both of these procs still show the `to_chat()` message even if it wasn't powered and nothing changed, so the second part should probably be indented.
```suggestion
/obj/machinery/atmospherics/proc/toggle(mob/living/user)
	if(powered())
		on = !on
		update_icon()
		if(user)
			to_chat(user, "<span class='notice'>You toggle [src] [on ? "on" : "off"].</span>")
```

File: code/modules/atmospherics/machinery/atmospherics.dm [hunk 4]
```diff
@@ -353,3 +356,33 @@ Pipelines + Other Objects -> Pipe network
...
+ *
+ * Arguments:
+ * * user - the mob who is setting the output pressure to maximum.
+ */
+/obj/machinery/atmospherics/proc/set_max(mob/living/user)
+	if(powered())
+		target_pressure = MAX_OUTPUT_PRESSURE
+		update_icon()
+	if(user)
+		to_chat(user, "<span class='notice'>You set the target pressure of [src] to maximum.</span>")
```
Comment (SabreML) [inline code/modules/atmospherics/machinery/atmospherics.dm:388]:
```suggestion
/obj/machinery/atmospherics/proc/set_max(mob/living/user)
	if(powered())
		target_pressure = MAX_OUTPUT_PRESSURE
		update_icon()
		if(user)
			to_chat(user, "<span class='notice'>You set the target pressure of [src] to maximum.</span>")
```

File: code/modules/atmospherics/machinery/atmospherics.dm [hunk 5]
```diff
@@ -353,3 +356,33 @@ Pipelines + Other Objects -> Pipe network
...
+ * Maxes the output pressure of the machine. If this is done by a user, display a message to them.
+ *
+ * NOTE: Only applies to atmospherics machines which allow a `target_pressure` to be set, such as pumps, or other devices.
+ *
+ * Arguments:
+ * * user - the mob who is setting the output pressure to maximum.
+ */
+/obj/machinery/atmospherics/proc/set_max(mob/living/user)
+	if(powered())
+		target_pressure = MAX_OUTPUT_PRESSURE
```
Comment (SabreML) [inline code/modules/atmospherics/machinery/atmospherics.dm:385]: No shortcut to set it to the minimum?

File: code/modules/atmospherics/machinery/components/binary_devices/pump.dm [hunk 6]
```diff
@@ -21,51 +21,31 @@ Thus, the two variables affect pump operation are set in New():
...
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	set_max()
-	return
+	if(can_use_shortcut(user))
+		set_max(user)
 
 /obj/machinery/atmospherics/binary/pump/AIAltClick()
 	set_max()
 	return ..()
```
Comment (SabreML) [inline code/modules/atmospherics/machinery/components/binary_devices/pump.dm:47]: Because of this line in `ai.dm`, if the AI alt clicks on a pump `set_max()` will be called twice. Once with a `user` argument and once without:
https://github.com/ParadiseSS13/Paradise/blob/fe6e445760ade9158e5fcf84346b11d1cfb15d42/code/_onclick/ai.dm#L159-L160
I'd recommend just changing that to just `return` like all of the other procs there, but since it was added in 21a9d929304cc8c742369447c0c443028e879c09 7 years ago it might be tied into some other interactions, so probably best to leave that for another PR unless you can make sure there won't be any side effects.

TLDR: You can get rid of the `return ..()` and add a user argument here and on the others to make it work properly, but since it's only setting it to maximum it doesn't actually do anything bad.
```suggestion
/obj/machinery/atmospherics/binary/pump/AIAltClick(mob/user)
	set_max(user)
```

File: code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm [hunk 7]
```diff
@@ -21,51 +21,31 @@ Thus, the two variables affect pump operation are set in New():
...
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	set_max()
-	return
+	if(can_use_shortcut(user))
+		set_max(user)
 
 /obj/machinery/atmospherics/binary/volume_pump/AIAltClick()
 	set_max()
 	return ..()
```
Comment (SabreML) [inline code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm:47]: Same issue here, called twice by AIs.
```suggestion
/obj/machinery/atmospherics/binary/volume_pump/AIAltClick(mob/user)
	set_max(user)
```

File: code/modules/atmospherics/machinery/components/trinary_devices/filter.dm [hunk 8]
```diff
@@ -31,46 +31,27 @@
...
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	set_max()
-	return
+	if(can_use_shortcut(user))
+		set_max(user)
 
 /obj/machinery/atmospherics/trinary/filter/AIAltClick()
 	set_max()
 	return ..()
```
Comment (SabreML) [inline code/modules/atmospherics/machinery/components/trinary_devices/filter.dm:53]:
```suggestion
/obj/machinery/atmospherics/trinary/filter/AIAltClick(mob/user)
	set_max(user)
```

File: code/modules/atmospherics/machinery/components/trinary_devices/mixer.dm [hunk 9]
```diff
@@ -6,37 +6,28 @@
...
-		return
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	set_max()
-	return
+	if(can_use_shortcut(user))
+		set_max(user)
 
 /obj/machinery/atmospherics/trinary/mixer/AIAltClick()
 	set_max()
```
Comment (SabreML) [inline code/modules/atmospherics/machinery/components/trinary_devices/mixer.dm:33]: Can't select this one properly to suggest changes, but you get the idea.

File: code/modules/atmospherics/machinery/atmospherics.dm (hunk 5 above)
Comment (SteelSlayer) [reply code/modules/atmospherics/machinery/atmospherics.dm:385]: Control click and alt click are already taken so I wouldn't be able.

Comment (SabreML) [reply code/modules/atmospherics/machinery/atmospherics.dm:385]: Oh true, fair enough.

File: code/game/machinery/machinery.dm (hunk 2 above)
Comment (SteelSlayer) [reply code/game/machinery/machinery.dm:665]: Yeah my logic was crap here. I'm rewriting it a bit differently than you suggested though.
```dm
/obj/machinery/proc/can_use_shortcut(mob/living/user)
	if(user.incapacitated())
		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
		return FALSE
	if(ishuman(user) && in_range(src, user))
		return TRUE
	if(issilicon(user))
		return TRUE
	return FALSE

```

File: code/modules/atmospherics/machinery/components/binary_devices/pump.dm (hunk 6 above)
Comment (SteelSlayer) [reply code/modules/atmospherics/machinery/components/binary_devices/pump.dm:47]: Yeah good catche on this stuff.

**Review by variableundefined — CHANGES_REQUESTED** [review]
DM on discord when deconflicted with link for merge.

**Review by variableundefined — APPROVED** [review]
